src/kakarot/stack.cairo

```diff
@@ -50,8 +50,10 @@ namespace Stack {
     // @param element The element to push.
     // @return stack The new pointer to the stack.
     func push_uint128{stack: model.Stack*}(element: felt) {
-        tempvar item = new Uint256(element, 0);
-        push(item);
+        tempvar item = Uint256(element, 0);
+        let (ap_val) = get_ap();
+        let item_ptr = cast(ap_val - 2, Uint256*);
+        push(item_ptr);
```

Comment:
The original implementation with `new` is 50 steps
The current one with `get_ap` is 49 steps 
and the following ones with `locals` is 47
```
func push_uint128{stack: model.Stack*}(element: felt) {
    alloc_locals;
    local item: Uint256 = Uint256(element, 0);
    let (__fp__, _) = get_fp_and_pc();
    push(&item);
    return ();
}
```
Test run 
`poetry run pytest tests/src/kakarot -k test_should_add_an_element_to_the_stack --profile-cairo  `

Reply:
let's use the local version then!

Reply:
done with 6c83ff0

as predicted the improvements are really low  141773 steps with ap_version to 141714 with locals
![profile004](https://github.com/user-attachments/assets/24e3355a-34f6-4703-b34b-1120ee3ae940)